### DISCUSSION TITLE
Data separation for teams - how can data be seperated in one Supabase project?

### DISCUSSION THREAD

**Discussion by torgem** [supabase/supabase#12672] [Questions]
I read a lot about the Supabase/PostgreSQL feature "row based security".
Is there a way / best practice to seperate data alternatively, e. g. by schema?
Background: The application which we are looking for a database backend for will be used by different teams and the data of that teams has to be strictly separated. My first thought was to use schemas to separate the data. The data structure would be the same. So in the end there are n separated data "silos" with the data for each team. A user ist assigned to 1 - n teams, so the permission to access the data "silos" depends on the teams assigned.

Example: 10 Silos with same data structure (tables, functions, ...). 10 Teams (each team has one silo). User A is assigned to Team 1 and 2 -> has access to data silo 1 and 2, but cannot access silo 3 to 10. The user can choose/change the team/silo at the front end.

Is supabase a possible solution here? Is there a way to restrict a user to 1 - n schemas based on configuration? The API calls should then fail when trying to access a schema/data silo which ist not assigned to the user's teams.

**Comment by steve-chavez** [comment]
>  Is there a way to restrict a user to 1 - n schemas based on configuration?

How many users will you have in your setup? If they're few, it might be possible to use [custom roles](https://github.com/supabase/supabase/discussions/11948#discussioncomment-4802137) for this and just do `GRANT USAGE ON SCHEMA <foo> TO role1`.
